src/smart-components/role/RolesTable.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useDataViewSelection, useDataViewPagination } from '@patternfly/react-data-view/dist/dynamic/Hooks';
+import { BulkSelect, BulkSelectValue } from '@patternfly/react-component-groups/dist/dynamic/BulkSelect';
+import { DataView } from '@patternfly/react-data-view/dist/dynamic/DataView';
+import { DataViewToolbar } from '@patternfly/react-data-view/dist/dynamic/DataViewToolbar';
+import { DataViewTable } from '@patternfly/react-data-view/dist/dynamic/DataViewTable';
+import { PageSection, Pagination } from '@patternfly/react-core';
+import { ActionsColumn } from '@patternfly/react-table';
+import ContentHeader from '@patternfly/react-component-groups/dist/esm/ContentHeader';
+import { fetchRoles } from '../../redux/actions/role-actions';
+import { useIntl } from 'react-intl';
+import messages from '../../Messages';
+import { mappedProps } from '../../helpers/shared/helpers';
+import { Role } from '../../redux/reducers/role-reducer';
+import { RBACStore } from '../../redux/store';
+
+const intl = useIntl();
```

Comment:
@InsaneZein here you are using the hook outside the component. It's probably the cause of the error

Reply:
Thank you! I knew I missed one somewhere :sweat_smile: